Makefile

```diff
@@ -0,0 +1,132 @@
+# Output registry and image names for operator image
+# Set env to override this value
+REGISTRY ?= openebs
+
+# Output plugin name and its image name and tag
+PLUGIN_NAME=jiva-csi
+PLUGIN_TAG=ci
+
+# Tools required for different make targets or for development purposes
+EXTERNAL_TOOLS=\
+	golang.org/x/tools/cmd/cover \
+	github.com/axw/gocov/gocov \
+	github.com/ugorji/go/codec/codecgen \
+
+# Lint our code. Reference: https://golang.org/cmd/vet/
+VETARGS?=-asmdecl -atomic -bool -buildtags -copylocks -methods \
+         -nilfunc -printf -rangeloops -shift -structtag -unsafeptr
+
+GIT_BRANCH = $(shell git rev-parse --abbrev-ref HEAD | sed -e "s/.*\\///")
+GIT_TAG = $(shell git describe --tags)
+
+# use git branch as default version if not set by env variable, if HEAD is detached that use the most recent tag
+VERSION ?= $(if $(subst HEAD,,${GIT_BRANCH}),$(GIT_BRANCH),$(GIT_TAG))
+COMMIT ?= $(shell git rev-parse HEAD | cut -c 1-7)
+
+ifeq ($(GIT_TAG),)
+	GIT_TAG := $(COMMIT)
+endif
+
+ifeq (${TRAVIS_TAG}, )
+  GIT_TAG = $(COMMIT)
+	export GIT_TAG
+else
+  GIT_TAG = ${TRAVIS_TAG}
+	export GIT_TAG
+endif
+
+PACKAGES = $(shell go list ./... | grep -v 'vendor')
+
+DATETIME ?= $(shell date +'%F_%T')
+LDFLAGS ?= \
+        -extldflags "-static" \
+	-X github.com/openebs/jiva-csi/version/version.Version=${VERSION} \
+	-X github.com/openebs/jiva-csi/version/version.Commit=${COMMIT} \
+	-X github.com/openebs/jiva-csi/version/version.DateTime=${DATETIME}
+
+
+.PHONY: all
+all:
```

Comment:
make all, generally builds everything, Here we are printing the help only. Can we have make help here. And make all will make everything (vet, fmt, build, image etc)?